tests/unit/test_diagnostics.py

```diff
@@ -0,0 +1,166 @@
+
+    def test_bignk(self, seed=0):
+        """Identifying the optimal summary statistics combination following the bivariate-g-and-k model.
+
+        Testing the Two-Stage Procedure's implementation for 2-D data.
+
+        Parameters
+        ----------
+        seed : int, optional
+
+        """
+        # Defining summary statistics.
+        ss_order = GNK.ss_order
+        ss_robust = GNK.ss_robust
+        ss_octile = GNK.ss_octile
+
+        list_ss = [ss_order, ss_robust, ss_octile]
+
+        # Initialising the simulator.
+        true_params = [3, 4, 1, 0.5, 1, 2, .5, .4, 0.6]
+        priors = []
+        priors.append(elfi.Prior('uniform', 0, 5, name='A1'))
+        priors.append(elfi.Prior('uniform', 0, 5, name='A2'))
+        priors.append(elfi.Prior('uniform', 0, 5, name='B1'))
+        priors.append(elfi.Prior('uniform', 0, 5, name='B2'))
+        priors.append(elfi.Prior('uniform', -5, 10, name='g1'))
+        priors.append(elfi.Prior('uniform', -5, 10, name='g2'))
+        priors.append(elfi.Prior('uniform', -.5, 5.5, name='k1'))
+        priors.append(elfi.Prior('uniform', -.5, 5.5, name='k2'))
+        EPS = np.finfo(float).eps
+        priors.append(elfi.Prior('uniform', -1 + EPS, 2 - 2 * EPS, name='rho'))
+
+        fn_simulator = BiGNK.BiGNK
+        y_obs = fn_simulator(*true_params, random_state=np.random.RandomState(seed))
+        simulator = elfi.Simulator(fn_simulator, *priors, observed=y_obs)
+
+        # Identifying the optimal summary statistics based on the Two-Stage procedure.
+        diagnostics = TwoStageSelection(simulator, euclidean_multiss, list_ss=list_ss, seed=seed)
+        diagnostics.run(n_sim=1000, n_acc=100, n_closest=20)
+
+    def test_gnk(self, seed=0):
+        """Identifying the optimal summary statistics combination following the g-and-k model.
+
+        Testing the Two-Stage Procedure's implementation for 1-D data.
+
+        Parameters
+        ----------
+        seed : int, optional
+
+        """
+        # Defining summary statistics.
+        ss_order = GNK.ss_order
+        ss_robust = GNK.ss_robust
+        ss_octile = GNK.ss_octile
+
+        list_ss = [ss_order, ss_robust, ss_octile]
+
+        # Initialising the simulator.
+        true_params = [3, 1, 2, .5]
+        priors = []
+        priors.append(elfi.Prior('uniform', 0, 10, name='A'))
+        priors.append(elfi.Prior('uniform', 0, 10, name='B'))
+        priors.append(elfi.Prior('uniform', 0, 10, name='g'))
+        priors.append(elfi.Prior('uniform', 0, 10, name='k'))
+
+        fn_simulator = GNK.GNK
+        y_obs = fn_simulator(*true_params, random_state=np.random.RandomState(seed))
+        simulator = elfi.Simulator(fn_simulator, *priors, observed=y_obs)
+
+        # Identifying the optimal summary statistics based on the Two-Stage procedure.
+        diagnostics = TwoStageSelection(simulator, euclidean_multiss, list_ss=list_ss, seed=seed)
+        diagnostics.run(n_sim=1000, n_acc=100, n_closest=20)
```

Comment:
`assert` something?

Though, could these 4 tests be combined? How long does it take to run them?

Reply:
They take 40 seconds to run.

Combining them would reduce readability, we could remove some of them or mark as slow if you don't see a purpose of having them.